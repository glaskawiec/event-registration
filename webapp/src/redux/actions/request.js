import { apiUri } from '../../config/default.json';

import {
    REQUEST_CREATE_EVENT_PENDING,
    REQUEST_CREATE_EVENT_SUCCESS,
    REQUEST_CREATE_EVENT_FAILURE,
    REQUEST_CREATE_EVENT_RESET
} from '../types/request'

export const createEventPending = () => ({
    type: REQUEST_CREATE_EVENT_PENDING
})

export const createEventSuccess = () => ({
    type: REQUEST_CREATE_EVENT_SUCCESS
})

export const createEventFailure = () => ({
    type: REQUEST_CREATE_EVENT_FAILURE
})

export const createEventReset = () => ({
    type: REQUEST_CREATE_EVENT_RESET
})

export const createEvent = (event) => {
    return dispatch => {
        dispatch(createEventPending());
        return fetch(`${apiUri}/events`, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(event)
        })
            .then((response) => {
                if (response.ok) {
                    dispatch(createEventSuccess())
                } else {
                    dispatch(createEventFailure())
                }
            })
            .catch(() => { dispatch(createEventFailure()) });
    }
}

