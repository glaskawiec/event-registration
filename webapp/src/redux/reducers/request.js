import {
    REQUEST_CREATE_EVENT_PENDING,
    REQUEST_CREATE_EVENT_SUCCESS,
    REQUEST_CREATE_EVENT_FAILURE,
    REQUEST_CREATE_EVENT_RESET
} from '../types/request'

export const initialState = {
    createEvent: {
        isPending: false,
        hadError: false,
        isSucceeded: false,
    }
}

const requestReducer = (state = initialState, action) => {
    switch (action.type) {
        case REQUEST_CREATE_EVENT_PENDING:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isPending: true
                }
            }
        case REQUEST_CREATE_EVENT_SUCCESS:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isSucceeded: true,
                    isPending: false
                }
            }
        case REQUEST_CREATE_EVENT_FAILURE:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isPending: false,
                    hadError: true
                }
            }
        case REQUEST_CREATE_EVENT_RESET:
            return { ...initialState }
        default:
            return state
    }
}

export default requestReducer;