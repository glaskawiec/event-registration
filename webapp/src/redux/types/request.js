export const REQUEST_CREATE_EVENT_PENDING = 'REQUEST_CREATE_EVENT_PENDING'
export const REQUEST_CREATE_EVENT_SUCCESS = 'REQUEST_CREATE_EVENT_SUCCESS'
export const REQUEST_CREATE_EVENT_FAILURE = 'REQUEST_CREATE_EVENT_FAILURE'
export const REQUEST_CREATE_EVENT_RESET = 'REQUEST_CREATE_EVENT_RESET'