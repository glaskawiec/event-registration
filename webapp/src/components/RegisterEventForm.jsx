import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import { Form, Button, Alert } from 'react-bootstrap';
import DatePicker from './DatePicker';
import TextInput from './TextInput';
import { email, required } from '../utils/validations';
import { connect } from 'react-redux';
import { createEvent } from '../redux/actions/request';
import Moment from 'moment'
import momentLocalizer from 'react-widgets-moment';
export class RegisterEventForm extends React.Component {
  constructor(props) {
    super(props);
    Moment.locale('en');
    momentLocalizer();
  }

  onSubmit = (formData) => {
    const { dispatch } = this.props;
    dispatch(createEvent(formData));
  }

  render() {
    return (
      <Form onSubmit={this.props.handleSubmit(this.onSubmit)}>
        {
          [
            {
              id: 'firstName',
              label: 'First Name',
              component: TextInput,
              validate: required
            },
            {
              id: 'lastName',
              label: 'Last Name',
              component: TextInput,
              validate: required
            },
            {
              id: 'email',
              label: 'Email',
              component: TextInput,
              validate: [required, email]
            },
            {
              id: 'date',
              label: 'Date',
              component: DatePicker,
              validate: required
            }
          ].map((field) => (
            <Form.Group key={field.id}>
              <Form.Label>
                {field.label}
              </Form.Label>
              <Field
                name={field.id}
                label={field.label}
                component={field.component}
                validate={field.validate}
              />
            </Form.Group>
          ))}
        {
          this.props.hadError &&
          <Alert variant="danger">
            There was an error, please try again later.
          </Alert>
        }
        <Button
          type="submit"
          disabled={this.props.isPending} >
          {this.props.isPending ? 'Loadingâ€¦' : 'Register'}
        </Button>
      </Form>
    )
  }
}

RegisterEventForm.propTypes = {
  dispatch: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isPending: PropTypes.bool.isRequired,
  hadError: PropTypes.bool,
}

const ConnectedRegisterEventForm = connect(
  (state) => ({
    hadError: state.request.createEvent.hadError,
    isPending: state.request.createEvent.isPending,
  }),
  () => ({}),
)(RegisterEventForm)

export default reduxForm({
  form: 'createEvent',
})(ConnectedRegisterEventForm);
