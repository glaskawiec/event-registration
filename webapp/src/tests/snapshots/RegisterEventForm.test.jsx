import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import RegisterEventForm from '../../components/RegisterEventForm';
import { initialState as requestInitialState } from '../../redux/reducers/request';

const middlewares = []
const mockStore = configureStore(middlewares);
test('RegisterEventForm snapshot', () => {
    const initialState = {
        request: requestInitialState
    }
    const store = mockStore(initialState)
    const tree = renderer
        .create(
            <Provider store={store}>
                <RegisterEventForm />
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});