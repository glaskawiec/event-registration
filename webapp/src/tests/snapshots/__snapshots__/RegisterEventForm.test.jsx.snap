// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterEventForm snapshot 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      First Name
    </label>
    <input
      className="form-control"
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Last Name
    </label>
    <input
      className="form-control"
      name="lastName"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Email
    </label>
    <input
      className="form-control"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Date
    </label>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          autocomplete="nope"
          className="form-control"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          value=""
        />
      </div>
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Register
  </button>
</form>
`;
